src/database/DataTypes/PeriodSchedule.js

```diff
@@ -5,7 +5,24 @@
 
 import Realm from 'realm';
 
-export class PeriodSchedule extends Realm.Object {}
+export class PeriodSchedule extends Realm.Object {
+  getUseablePeriodsForProgram(program, maxMOS) {
+    const periods = this.periods.reduce(period => {
+      if (period.numberOfRequisitionsForProgram(program) >= maxMOS) return null;
+      return period;
+    });
+    return periods;
+  }
+
+  addPeriod(period) {
+    this.periods.push(period);
+  }
```

Comment:
Just a thought: this method is a one liner and likely will only be used in `addPeriodIfUnique`, which we'll usually (always?) want people to use. Just inline it in addPeriodIfUnique?

Reply:
cool, was following suit of the other models, removed